app/controllers/register/suitability/BeforeYouContinueController.scala

```diff
@@ -35,59 +35,64 @@ class BeforeYouContinueController @Inject()(
                                              appConfig: FrontendAppConfig
                                            )(implicit ec: ExecutionContext) extends FrontendBaseController with I18nSupport {
 
+  private def routeNonTaxable()(implicit request: MatchingAndSuitabilityDataRequest[_]) = if (appConfig.disableNonTaxableRegistrations) {
+    Redirect(controllers.register.suitability.routes.NoNeedToRegisterController.onPageLoad())
+  } else {
+    if (isAgentUser) {
+      Ok(nonTaxableAgentView())
+    } else {
+      Ok(nonTaxableView())
+    }
+  }
+
+  private def routeTaxable()(implicit request: MatchingAndSuitabilityDataRequest[_]) = if (isAgentUser) {
+    Ok(taxableAgentView())
+  } else {
+    Ok(taxableView())
+  }
+
   def onPageLoad(): Action[AnyContent] = actions.identifiedUserMatchingAndSuitabilityData() {
     implicit request =>
 
-      (doesTrustHaveUtr, isTrustTaxable) match {
-        case (Some(false), Some(true)) =>
-          Ok(taxableView())
-        case (Some(true), Some(true)) =>
-          Ok(existingTaxableView())
-        case (_, Some(false)) =>
-          if (appConfig.disableNonTaxableRegistrations) {
-            Redirect(controllers.register.suitability.routes.NoNeedToRegisterController.onPageLoad())
-          } else {
-            if (isAgentUser) {
-              Ok(nonTaxableAgentView())
-            } else {
-              Ok(nonTaxableView())
-            }
-          }
-        case _ =>
-          Redirect(controllers.register.routes.SessionExpiredController.onPageLoad())
+      if (isExistingTrust) {
+        Ok(existingTaxableView())
+      } else {
+        if (isTrustTaxable) {
+          routeTaxable()
+        } else {
+          routeNonTaxable()
+        }
       }
   }
 
   def onSubmit(): Action[AnyContent] = actions.identifiedUserMatchingAndSuitabilityData().async {
     implicit request =>
 
       featureFlagService.isNonTaxableAccessCodeEnabled() map { isNonTaxableAccessEnabled =>
-        Redirect {
-          (isExpressTrust, isTrustTaxable, isNonTaxableAccessEnabled) match {
-            case (Some(true), Some(false), true) =>
-              routes.NonTaxableTrustRegistrationAccessCodeController.onPageLoad().url
-            case _ =>
-              controllers.register.routes.CreateDraftRegistrationController.create().url
-          }
+        if(isExpressTrust && !isTrustTaxable && isNonTaxableAccessEnabled) {
+          Redirect(routes.NonTaxableTrustRegistrationAccessCodeController.onPageLoad().url)
+        } else {
+          Redirect(controllers.register.routes.CreateDraftRegistrationController.create().url)
         }
       }
   }
 
-  private def isTrustTaxable(implicit request: MatchingAndSuitabilityDataRequest[_]): Option[Boolean] =
-    request.getPage(TrustTaxableYesNoPage)
+  private def isTrustTaxable(implicit request: MatchingAndSuitabilityDataRequest[_]): Boolean =
+    request.getPage(TrustTaxableYesNoPage).getOrElse(true)
```

Comment:
I think these should be .contains(true) rather than .getOrElse(true/false)

Reply:
I was going with reasonable defaults as it's safe to assume they'll fit into either but I can do the contains to ensure it is there!